lifebit-ai/hic
# Fixes pipeline failure on cloudos

## Description

This PR fixes nf-core issues when running the nf-core/hic pipeline on CloudOS platform.

Lifebit for years used nextflow public pipelines developed by other people on CloudOS "curated pipelines" page, many of them from [nf-core community](https://nf-co.re/pipelines). Unfortunately, after some time the pipelines started failing with test data on CloudOS, which needs debugging. More than that, to ensure we don't encounter any future changes to pipelines that canbreak them on cloudOS, it was decided to keep a stable fork with fixed pipeline version for own lifebit use. This repository is the first one.

**Task for this PR:** make pipeline work on cloudos with test data (and ensure future-proof stability) by making as few changes to the repository as possible.

## Nf-core issues fixed to make the hic pipeline run on cloudos again:
 
- [x] **changed all instances of `$projectDir` to `$baseDir`**, because CloudOS has old nextflow version that doesn't support `$projectDir`

- [x] **added suport for `--config` parameter.** This is our workaround the obstacle that CloudOS doesn't allow to use nextflow's `-profile` parameter which is always used to define test parameters and profiles. With this new argument it is possible to get same outcome as with `-profile test` with `--config conf/test.config`

- [x] **set docker.enabled to be always true.** Removed necessity to specify docker profile (and its actually impossible on CloudOS, see above) by putting `docker.enabled = true` outside of any profile scope. So now hic pipeline always relies on docker, which is how actually it works on cloudos.

- [x] **fixed two output issues** where `/` is in the end of output folder name (apparently on NF v20.01.0 that fails)

- [x] **Re-Enabled ci tests** and removed one failing test. I just discovered that github actions are disabled in forks, so you have to manually re-enable them if you want tests work in your repo too.

- [x] **copied all test data from nf-core** github repository to Lifebit's s3 bucket. To make sure we won't ever lose them (that happened before)

## To test
Successful cloudOS run to check:
https://cloudos.lifebit.ai/app/jobs/60e7b2512f1720019afc5ce4 ([shareable link](https://cloudos.lifebit.ai/public/jobs/60e7b2512f1720019afc5ce4))
![image](https://user-images.githubusercontent.com/64809705/125047650-da350480-e0a7-11eb-9769-79e0cf94a773.png)

To run locally:
(will take ~4 minutes)
```
NXF_VER=20.01.0 nextflow run https://github.com/lifebit-ai/hic -r 82703058420a29db05e80b871fe97230e414b709 --config conf/test.config
```

## To Do:
Next steps are:
- to add a ci test to run test nextflow run directly on CloudOS (done - #2);
- to add a similar CloudOS ci test to a dedicated repository that will have such ci tests for all pipelines (done - https://github.com/lifebit-ai/ci-nextflow-tests/pull/1)
